spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -1189,91 +1191,48 @@ public InterpreterResult interpret(String[] lines, InterpreterContext context) {
     }
   }
 
-  public InterpreterResult interpretInput(String[] lines, InterpreterContext context) {
+  public InterpreterResult interpretInput(String code, InterpreterContext context) {
     SparkEnv.set(env);
 
-    String[] linesToRun = new String[lines.length];
-    for (int i = 0; i < lines.length; i++) {
-      linesToRun[i] = lines[i];
-    }
-
     Console.setOut(context.out);
-    out.setInterpreterOutput(context.out);
+    ByteArrayOutputStream replOutput = new ByteArrayOutputStream();
+    out.setInterpreterOutput(replOutput);
     context.out.clear();
-    Code r = null;
-    String incomplete = "";
-    boolean inComment = false;
-
-    for (int l = 0; l < linesToRun.length; l++) {
-      String s = linesToRun[l];
-      // check if next line starts with "." (but not ".." or "./") it is treated as an invocation
-      if (l + 1 < linesToRun.length) {
-        String nextLine = linesToRun[l + 1].trim();
-        boolean continuation = false;
-        if (nextLine.isEmpty()
-           || nextLine.startsWith("//")         // skip empty line or comment
-           || nextLine.startsWith("}")
-           || nextLine.startsWith("object")) {  // include "} object" for Scala companion object
```

Comment:
Do you have test for this scenario (Scala companion object) ? If it is broken, then we should either fix it or document it.